### DISCUSSION TITLE
Error while running `lerna run build`

### DISCUSSION THREAD

**Discussion by vitika9** [Atri-Labs/atrilabs-engine#224] [Help: Installation & Start] [answered]
I followed the contributing guide but still getting this error
<img width="960" alt="image" src="https://user-images.githubusercontent.com/79760854/193426348-0abfe88a-b2ef-4335-a8c7-e1279f91eeee.png">

**Comment by anuragc10** [comment]
I am also getting the same error!!
Due to this, it is showing errors and warnings while running `yarn run start` command.
 @cruxcode, please help us.

OS: Windows (11)

**Reply by darshitac11** [reply]
@Vitika9 and @anuragc10 - Please run `lerna bootstrap` and `lerna run build` again. There is a known issue with `lerna bootstrap` whose workaround is that you run it a few times if all packages are not built in the first attempt.

We are planning to move away from `lerna` in the future versions to avoid this problem.

**Reply by anuragc10** [reply]
Sorry, but it is not helping, the same error is coming.

**Reply by vitika9** [reply]
Same here, not working.

**Reply by cruxcode** [reply]
Please run `git pull upstream main` and try again. This was a temporary issue with the typescript that has been resolved. We are sorry for the inconvenience. I am assuming that you have already setup upstream as instructed in CONTRIBUTING.md.

Also, please delete `packages/undo-redo-layer/node_modules/@atrilabs` directory if present.

**Reply by cruxcode** [reply]
Once you have run above, you should run `lerna run build` again.

**Reply by anuragc10** [reply]
Thanks for the help, @cruxcode @Vitika9 @darshitac11  , it's working properly.

**Comment by darshitac11** [comment, marked as answer]
1. Run `git pull upstream main` again. 
2. Delete `packages/undo-redo-layer/node_modules/@atrilabs` directory, if present.
3. Run `lerna run build` again.

**Comment by iamBijoyKar** [comment]
I followed the contributing guide but I am getting this error whenever I run `lerna run build`. Please help 

![image](https://user-images.githubusercontent.com/85790967/215110874-2f69c954-2277-48d0-ae8d-3b2936ae0c2b.png)

**Reply by cruxcode** [reply]
Did you run `yarn build` inside `packages/forest`, `packages/core` and `packages/scripts` before running `lerna run build`?

**Reply by iamBijoyKar** [reply]
Yes I did

**Reply by cruxcode** [reply]
Can you confirm that `packages/forest/lib` directory exists? If yes, then run `lerna run build` again.

**Reply by iamBijoyKar** [reply]
I tried to run `lerna run build` again but I am facing the same issue 

![image](https://user-images.githubusercontent.com/85790967/215323243-ee601fbb-00c9-49ec-9410-340058f0154a.png)

![image](https://user-images.githubusercontent.com/85790967/215323145-7b30fdf5-1f72-45f7-a156-94422c6fd407.png)

**Reply by cruxcode** [reply]
@iamBijoyKar we are doing a major update this week. After this update all this hassle of building packages will go away. I hope you don't mind waiting for a week.

**Reply by iamBijoyKar** [reply]
Thank you very much. I am looking forward to contribute in this project.
